### PR TITLE
Add formal state to onSubmit

### LINKED ISSUE #22: Pass formal object as a second argument to onSubmit
## The problem

Since `onSubmit` does not have yet access to the formal state, I may want to call the `formal` API on my submit function.

i.e.

```js
const formal = useFormal(initialValues, {
  onSubmit (values, formal) {
    console.log('your values:', values)
    formal.reset()
  }
})
```

## The solution

Pass the `formal` object as a second argument to `onSubmit`, here: https://github.com/kevinwolfcr/formal/blob/master/packages/formal/src/use-formal.ts#L86

## Notes

N/A

### PR DESCRIPTION
**Pull request by the-bionic**
# Whats new?

Pass formal state to `onSubmit` callback 

## Issue link

Closes #22

## Notes

What do you think? I could not unit-test it.

### CODE REVIEW

**Comment by kevinwolfcr** [comment]
Nice work, @the-bionic. I was thinking about how useful will be to pass the **whole** object to onSubmit. Do you think maybe we should pass like `setErrors`, `clearErrors` and `reset`? What will be the use cases of passing the whole object?

**Comment by the-bionic** [comment]
Yeah, I can't think of any use cases for them. I don't think there is any reason to check for those properties after submission. You'd typically be making an ajax request in there.

Every other check for `isDirty` or `isSubmitting` can happen inside the form.

**Comment by kevinwolfcr** [comment]
Ok so let's pass only `setErrors`, `clearErrors` and `reset`.

File: packages/formal/src/use-formal.ts [hunk 1]
```diff
@@ -114,6 +98,26 @@ export default function useFormal<Schema>(
     [errors, isDirty, isSubmitting, isValidating]
   )
 
+  const submit = useCallback(async () => {
```
Comment (kevinwolfcr) [inline packages/formal/src/use-formal.ts]: Can you please put this after `reset` method? Where it was previously.

**Pull request closed without merging**